clang/include/clang/AST/TypeLoc.h

```diff
@@ -1355,7 +1355,7 @@ class BlockPointerTypeLoc : public PointerLikeTypeLoc<BlockPointerTypeLoc,
 };
 
 struct MemberPointerLocInfo : public PointerLikeLocInfo {
-  TypeSourceInfo *ClassTInfo;
+  void *QualifierData = nullptr;
```

Comment:
Could we instead use one of LLVM's pointer union types? 

Ah, urgh, I see we're inheriting this from `NestedNameSpecifierLoc`.  Its a shame we don't have a better 'thing' to store.  I'm OK leaving a 'fixme' here instead.

Reply:
This is actually not just an issue with NestedSpecifierLoc.

TypeLoc provides access to the opaque data through this void pointer, through the getOpaqueData method.
I added a comment there, but chances are low somebody is going to do something about it anytime soon :)

There are far worse issues just on TypeLoc itself which are still there from the initial implementation 15 years ago.